src/components/ec-file-dropzone/ec-file-dropzone.vue

```diff
@@ -42,82 +42,86 @@
-    document.removeEventListener('dragover', this.onDragover);
-    document.removeEventListener('dragleave', this.onDragleave);
-    document.removeEventListener('drop', this.onDrop);
+const props = defineProps({
+  isDisabled: {
+    type: Boolean,
+    default: false,
   },
-  methods: {
-    onFileInputChange() {
-      const files = [...this.$refs.fileInput.files].filter(isNotFolder);
-      if (files.length) {
-        this.$emit('change', files);
-        this.$refs.fileInput.value = '';
-      }
-    },
-    onComponentDrop(dragEvent) {
-      if (!this.isDisabled) {
-        const files = [...dragEvent.dataTransfer.files].filter(isNotFolder);
-        if (files.length) {
-          this.$emit('change', files);
-        }
-      }
-      this.disableDragging();
-    },
-    onDrop(dragEvent) {
-      dragEvent.preventDefault();
-      this.disableDragging();
-    },
-    onDragenter() {
-      this.dragEnterLeaveEventCounter++;
-    },
-    onDragover(dragEvent) {
-      // Necessary for the drop listener to work:
-      // https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API#define_a_drop_zone
-      dragEvent.preventDefault();
-    },
-    onDragleave() {
-      this.dragEnterLeaveEventCounter--;
-    },
-    disableDragging() {
-      this.dragEnterLeaveEventCounter = 0;
-    },
-  },
-};
+});
+
+const dragEnterLeaveEventCounter = ref(0);
+const isNotFolder = file => !!file.type && !!file.size;
+const fileInput = ref(null);
+
+const isDragging = computed(() => dragEnterLeaveEventCounter.value > 0);
+
+onMounted(() => {
+  document.addEventListener('dragenter', onDragenter);
+  document.addEventListener('dragover', onDragover);
+  document.addEventListener('dragleave', onDragleave);
+  document.addEventListener('drop', onDrop);
+});
+
+onBeforeUnmount(() => {
+  document.removeEventListener('dragenter', onDragenter);
+  document.removeEventListener('dragover', onDragover);
+  document.removeEventListener('dragleave', onDragleave);
+  document.removeEventListener('drop', onDrop);
+});
+
+function onFileInputChange() {
+  const files = [...fileInput.value.files].filter(isNotFolder);
+  if (files.length) {
+    emit('change', files);
+    fileInput.value.value = '';
+  }
+}
+
+function onComponentDrop(dragEvent) {
+  if (!props.isDisabled) {
+    const files = [...dragEvent.dataTransfer.files].filter(isNotFolder);
+    if (files.length) {
+      emit('change', files);
+    }
+  }
+  disableDragging();
+}
+
+function onDrop(dragEvent) {
+  dragEvent.preventDefault();
+  disableDragging();
+}
+
+function onDragenter() {
+  dragEnterLeaveEventCounter.value++;
+}
+
+function onDragover(dragEvent) {
+  // Necessary for the drop listener to work:
+  // https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API#define_a_drop_zone
+  dragEvent.preventDefault();
+}
+
+function onDragleave() {
+  dragEnterLeaveEventCounter.value--;
+}
+
+function disableDragging() {
```

Comment:
The name of this function is not semantic, it not represents its goal. We can remove it and use isDragging in all the places it is used now.

Reply:
I don't think it should be removed, but I agree with the naming, maybe it can be renamed to resetDraggingCounter to accurately represent it.